internal/pkg/app/app_test.go

```diff
@@ -0,0 +1,80 @@
+package app
```

Comment:
This should be `package app_test`

Reply:
Do u mean moving the app_test.go in to /pkg/app_test? My unit tests does test private methods, but I would envision that there will be some tests around these methods. If we put them in separate packages, we will lose that, no?

Reply:
No:
* The file is called `app_test.go` and lives in `/app/` alongside `app.go`
* The package declared in `app_test.go` is `package app_test`
* The code in the file tests out the public interface of the app package.  In particular, for the things defined in `app.go`

I guess the crux of the matter is black (public API) versus white (internal function) box texting.  It seems like most resources say to focus on testing of the public API.  This result of this is that:

* The API is less likely to change and therefore the tests are less brittle
* This pattern helps prevent dependency cycles when trying to test code

For reference:
* https://codefarm.me/2017/07/01/testing-in-go-language/#white-box-testing
* https://github.com/shekhargulati/52-technologies-in-2016/blob/master/29-go-unit-testing/README.md#black-box-testing-vs-white-box-testing
* https://stackoverflow.com/questions/19998250/proper-package-naming-for-testing-with-the-go-language/31443271#31443271
* https://speakerdeck.com/mitchellh/advanced-testing-with-go?slide=34
* https://talks.golang.org/2014/testing.slide#4, https://talks.golang.org/2014/testing.slide#20, https://talks.golang.org/2014/testing.slide#21

We need to decide what the expectation/pattern is going to be across the code base.

Reply:
Ok, I am fine with updating package to `app_test` since the unit tests does not access any internal methods. If we need to do whitebox unit tests later (which I think can be valuable), we can have a separate file to test them out.